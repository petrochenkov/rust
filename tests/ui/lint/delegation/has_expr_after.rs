#![deny(delegations_detailed, delegations_per_parent_stats)]

struct A;

impl A {
    fn foo(&self) -> i32 { 0 }
    fn bar(&self) -> Self { A }
}

struct B(A);

impl B {
    fn foo(&self) {
        self.0.foo().clone() as i32;
        //~^ ERROR dstats. stmts: OneWithoutTail, args_match: Same, ret_match: Different, callee_has_self: true, caller_has_self: true, same_name: false, has_expr_after: true.
        //~| ERROR dstats. stmts: OneWithoutTail, args_match: Same, ret_match: Different, callee_has_self: true, caller_has_self: true, same_name: true, has_expr_after: true.
    }

    fn bar(&self) -> Self {
        Self(self.0.bar())
        //~^ ERROR dstats. stmts: ZeroWithTail, args_match: Same, ret_match: Same, callee_has_self: false, caller_has_self: true, same_name: false, has_expr_after: false.
        //~| ERROR dstats. stmts: ZeroWithTail, args_match: Same, ret_match: SameUpToSelfType, callee_has_self: true, caller_has_self: true, same_name: true, has_expr_after: true.
    }
}

fn main() {}
