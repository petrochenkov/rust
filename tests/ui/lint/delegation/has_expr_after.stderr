error: dstats. stmts: OneWithoutTail, args_match: Same, ret_match: Different, self_arg: Value, same_name: false, has_expr_after: true.
  --> $DIR/has_expr_after.rs:14:9
   |
LL | /     fn foo(&self) {
LL | |         self.0.foo().clone() as i32;
   | |         ^^^^^^^^^^^^^^^^^^^^ callee
LL | |
LL | |
LL | |     }
   | |_____- caller
   |
note: the lint level is defined here
  --> $DIR/has_expr_after.rs:1:9
   |
LL | #![deny(delegations_detailed, delegations_per_parent_stats)]
   |         ^^^^^^^^^^^^^^^^^^^^

error: dstats. stmts: OneWithoutTail, args_match: Same, ret_match: Different, self_arg: Value, same_name: true, has_expr_after: true.
  --> $DIR/has_expr_after.rs:14:9
   |
LL | /     fn foo(&self) {
LL | |         self.0.foo().clone() as i32;
   | |         ^^^^^^^^^^^^ callee
LL | |
LL | |
LL | |     }
   | |_____- caller

error: dstats. stmts: ZeroWithTail, args_match: Different, ret_match: Same, self_arg: Type, same_name: false, has_expr_after: false.
  --> $DIR/has_expr_after.rs:20:9
   |
LL | /     fn bar(&self) -> Self {
LL | |         Self(self.0.bar())
   | |         ^^^^^^^^^^^^^^^^^^ callee
LL | |
LL | |
LL | |     }
   | |_____- caller

error: dstats. stmts: ZeroWithTail, args_match: Same, ret_match: SameUpToSelfType, self_arg: Value, same_name: true, has_expr_after: true.
  --> $DIR/has_expr_after.rs:20:14
   |
LL | /     fn bar(&self) -> Self {
LL | |         Self(self.0.bar())
   | |              ^^^^^^^^^^^^ callee
LL | |
LL | |
LL | |     }
   | |_____- caller

error: delegation methods stats. methods_count: 2, 1.
   |
note: the lint level is defined here
  --> $DIR/has_expr_after.rs:1:31
   |
LL | #![deny(delegations_detailed, delegations_per_parent_stats)]
   |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 5 previous errors

